pkgs/shells/fish/default.nix

```diff
@@ -141,9 +148,9 @@ let
     end # fenv
   '';
 
-  fish = stdenv.mkDerivation rec {
+  fish = stdenv.mkDerivation (finalAttrs: {
     pname = "fish";
-    version = "3.7.1";
+    version = "4.0b1";
 
     src = fetchurl {
```

Comment:
Can't you use a `fetchFromGitHub`?

Reply:
Did you read the comment just below, telling you why FFGH isn't used but instead the release tarball is fetched manually?

Though as this is a major release, we should verify if this difference still exists between source code from the repo and released source code.

Reply:
I think this is a good opportunity to switch the ffgh for better security. The fish version can be set as an env var https://codeberg.org/adamcstephens/nix-sandbox/src/commit/ad75d616bebe5b52a2178bc24151b500846b41e4/packages/fish/package.nix#L39

Reply:
If the documentation note still exists we should definitely investigate that, but I’d feel much more comfortable using the git source.

Reply:
Yeah, I'll have a look at that

Reply:
If you fetch from github, you'll have to generate the docs yourself, which needs sphinx.

Reply:
Right, I'll have to check how that's done then.

Reply:
In terms of upstream: Install sphinx before you run cmake.

No idea if nix needs anything special.

Reply:
> Install sphinx before you run cmake.

Also known as just putting `sphinx` in `nativeBuildInputs` and letting `cmake` pick it up.

(ideally put the `fetchFromGithub` and `sphinx` improvements in separate commits.

Reply:
There was a bit more to it. I ran into an issue with fish_indent_lexer, and I'm not sure why upstream didn't run into this. I'll send the patch upstream in any case to see what they say.